### PR TITLE
Make E2E test only use the GUI, drop Mysqli connection

### PR DESCRIPTION
**Pull request by williamdes**
Goal: test anything, anywhere.

```
yarn global add selenium-standalone
~/.yarn/bin/selenium-standalone install
~/.yarn/bin/selenium-standalone start --  -port 4444
```

```
TESTSUITE_DATABASE_PREFIX="e2epma" \
TESTSUITE_SELENIUM_HOST="127.0.0.1" TESTSUITE_SELENIUM_PORT="4444" TESTSUITE_USER="root" \
TESTSUITE_URL="https://demo.phpmyadmin.net/master/" \
TESTSUITE_PASSWORD="" ./vendor/bin/phpunit --group=selenium --no-coverage
```

#### Docker way

```
docker run -d -e SCREEN_WIDTH=1920 -e SCREEN_HEIGHT=1080 --rm --name=selenium --net=host -v /dev/shm:/dev/shm selenium/standalone-chrome:3.141.59
```

Related to https://github.com/phpmyadmin/phpmyadmin/pull/15462 and https://github.com/phpmyadmin/phpmyadmin/issues/16125

### CODE REVIEW

**Comment by williamdes** [comment]
@mauriciofauth `Table\StructureTest::testDropColumns` is broken the code says error 404

**Comment by williamdes** [comment]
https://github.com/phpmyadmin/phpmyadmin/wiki/Unit_Testing/5bbf3ac4e849391a7d81608d94d55b183802de40

**Comment by mauriciofauth** [comment]
> @mauriciofauth `Table\StructureTest::testDropColumns` is broken the code says error 404

I was not able to reproduce that.

**Comment by williamdes** [comment]
@mauriciofauth I still can reproduce this, this time on latest master commit
image

**Comment by mauriciofauth** [comment]
> @mauriciofauth I still can reproduce this, this time on latest master commit
> image
> 

Is it only with the drop button? The other actions are working?

**Comment by mauriciofauth** [comment]
Did you cleared the routes cache?

**Comment by williamdes** [comment]
> Did you cleared the routes cache?

Yes, It did not fix the issue
You can reproduce it on demo server at https://demo.phpmyadmin.net/master/index.php?route=/table/structure&db=world&table=Country

**Review by MauricioFauth — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -201,9 +200,10 @@ jobs:
         - TESTSUITE_URL=http://127.0.0.1:8000
         - TESTSUITE_SERVER="127.0.0.1"
         - TESTSUITE_DATABASE="selenium"
+        - TESTSUITE_SELENIUM_HOST="127.0.0.1"
+        - TESTSUITE_SELENIUM_PORT="4444"
       before_script:
         - export TESTSUITE_PASSWORD=`openssl rand -base64 30`
-        - export TESTSUITE_BROWSERSTACK_KEY=`echo cHlDcHJTNmZwZjVlaUR2RmV6VkU= | base64 --decode`
```
Comment (williamdes) [inline .travis.yml:206]: @mauriciofauth are you okay with using a driver in the CI and dropping BrowserStack ?

**Comment by williamdes** [comment]
I think this PR is ready for merging if everyone agrees to drop BrowserStack

Only 8 errors: https://travis-ci.org/github/williamdes/phpmyadmintest/jobs/690740323#L583 (good results for a first run)

About the duration:
- Local docker setup: ~20m
- BrowserStack before this PR: ~30m (~ 30s / test)
- BrowserStack after this PR: ~60m (1h) (~1 min / test)

The difference is because we use the GUI to do everything and having 2 tabs takes more time obviousy.

The TravisCI time limit is 50min, I am forced either to abandon my work or to drop BrowserStack

Additional notes:
- The scripts are now better and more contributor friendly.
- Using docker image as they are the best stable option nowdays.
- The selenium tests will run on forks !

**Review by devenbansod — COMMENTED** [review]
Great work @williamdes! 💯 🥇 This looks good to me overall. 

I think we started using Browserstack because along with providing GUI instances on multiple OSs, browsers and mobile screens, they provided better debugging info (ex. video recordings, screenshots etc.). But a huge issue was inability to run the tests on forks.

If Browserstack can co-exist with these changes, we can probably run the browserstack version of the tests (possibly on multiple OSs and browsers) as a part of nightly build and run the driver-based tests otherwise?

**Comment by williamdes** [comment]
thank you!

> If Browserstack can co-exist with these changes, we can probably run the browserstack version of the tests (possibly on multiple OSs and browsers) as a part of nightly build and run the driver-based tests otherwise?

Time is the issue, we can only keep the ability to run individual tests. 

Remote proxies testing is cool by the features they have but incompatible with additional tests or more complex ones as the time will reach 50 minutes very rapidly :)

**Comment by mauriciofauth** [comment]
One thing I thought some time ago was to test more things in each test to reduce the time spent in the test setup. For example, for `Table\StructureTest` we have three tests, we can reduce it to just one and tests everything in table structure page in one test. I think this will help reducing the total time.

File: .travis.yml (hunk 1 above)
Comment (mauriciofauth) [reply .travis.yml:206]: You replaced BrowserStack with a Docker image that runs inside the CI server, right? I'm okay with that. I'll miss the BrowserStack videos though.

**Comment by williamdes** [comment]
> One thing I thought some time ago was to test more things in each test to reduce the time spent in the test setup. For example, for `Table\StructureTest` we have three tests, we can reduce it to just one and test everything in table structure page in one test. I think this will help reducing the total time.

I agree partially, this would be a solution but the code quality would be worse
This is more a patch solution IMO

File: .travis.yml (hunk 1 above)
Comment (williamdes) [reply .travis.yml:206]: Yes, right.
I miss them too

**Comment by williamdes** [comment]
```
docker run -d -e SCREEN_WIDTH=1920 -e SCREEN_HEIGHT=1080 -e VNC_NO_PASSWORD=1 --rm --name=selenium --net=host -v /dev/shm:/dev/shm selenium/standalone-chrome-debug:3.141.59
```

This will start a VNC server with the selenium server (on localhost because of `--net=host`)
You can connect and have a visual on what is going on.

**Comment by williamdes** [comment]
One last rebase, all tests are green and stable.
Merging soon :tada:

**Pull request merged**